src/scripts/get-wpt-revision.py

```diff
@@ -32,17 +37,42 @@ def request(method, url):
     conn.close()
 
 
-def main(interval):
+@contextlib.contextmanager
+def get_with_retry(url, interval, timeout):
+    try:
+        with request('GET', url) as response:
+            status = response.status
+            message = 'HTTP Error %s: %s' % (status, response.reason)
+
+            if status >= 200 and status < 300:
+                yield response
+                return
+    except Exception as exception:
+        message = str(exception)
+
+    remaining = timeout - interval
+
+    if remaining <= 0:
+        logger.warn('Retry timeout exceeded.')
+        raise Exception(message)
+
+    logging.warn(message)
+    logging.warn('Retrying in %s seconds...' % (interval,))
+
+    time.sleep(interval)
+
+    with request_with_retry(method, url, interval, remaining) as response:
```

Comment:
Is this magic naming that will delegate to `get_with_retry` on `method == 'GET'`? Is `method` a global? I don't see it anywhere else in the `get_with_retry` body.

Reply:
This is a typo I introduced just before submitting the patch. Good catch!